src/sinks/aws_kinesis_firehose.rs

```diff
@@ -0,0 +1,374 @@
+use crate::{
+    dns::Resolver,
+    event::{self, Event},
+    region::RegionOrEndpoint,
+    sinks::util::{retries::RetryLogic, BatchConfig, SinkExt, TowerRequestConfig},
+    topology::config::{DataType, SinkConfig, SinkContext, SinkDescription},
+};
+use bytes::Bytes;
+use futures::{stream::iter_ok, Future, Poll, Sink};
+use lazy_static::lazy_static;
+use rusoto_core::{Region, RusotoError, RusotoFuture};
+use rusoto_firehose::{
+    KinesisFirehose, KinesisFirehoseClient, ListDeliveryStreamsInput, PutRecordBatchError,
+    PutRecordBatchInput, PutRecordBatchOutput, Record,
+};
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+use std::{convert::TryInto, fmt, sync::Arc};
+use tower::Service;
+use tracing_futures::{Instrument, Instrumented};
+
+#[derive(Clone)]
+pub struct KinesisFirehoseService {
+    client: Arc<KinesisFirehoseClient>,
```

Comment:
Don't think we need to arc this https://docs.rs/rusoto_firehose/0.42.0/rusoto_firehose/struct.KinesisFirehoseClient.html#impl-Clone